sbin/build.sh

```diff
@@ -22,211 +22,260 @@ OPENJDK_REPO_NAME=$3
 
-FOUND_FREETYPE=$(find "$WORKING_DIR/$OPENJDK_REPO_NAME/installedfreetype/lib" -name "libfreetype.so.6.5.0")
+  if [[ ! -z "$FOUND_ALSA" ]] ; then
+    echo "Skipping ALSA download"
+  else
+    wget -nc ftp://ftp.alsa-project.org/pub/lib/alsa-lib-1.0.27.2.tar.bz2
+    tar xvf alsa-lib-1.0.27.2.tar.bz2
+    rm alsa-lib-1.0.27.2.tar.bz2
+  fi
+}
 
-if [[ ! -z $FOUND_FREETYPE ]] ; then
-  echo "Skipping FreeType download"
-else
-  # Then FreeType for fonts: make it and use
-  wget -nc http://ftp.acc.umu.se/mirror/gnu.org/savannah/freetype/freetype-2.4.0.tar.gz
-   
-  tar xvf freetype-2.4.0.tar.gz
-  rm freetype-2.4.0.tar.gz
+checkingAndDownloadingFreetype()
+{
+  echo "Checking for freetype"
 
-  cd freetype-2.4.0
+  FOUND_FREETYPE=$(find "$WORKING_DIR/$OPENJDK_REPO_NAME/installedfreetype/lib" -name "libfreetype.so.6.5.0")
 
-  if [ $(uname -m) = "ppc64le" ]; then
-    PARAMS="--build=$(rpm --eval %{_host})"
-  fi
-   
-  # We get the files we need at $WORKING_DIR/installedfreetype
-  bash ./configure --prefix=$WORKING_DIR/$OPENJDK_REPO_NAME/installedfreetype $PARAMS && make all && make install
+  if [[ ! -z "$FOUND_FREETYPE" ]] ; then
+    echo "Skipping FreeType download"
+  else
+    # Then FreeType for fonts: make it and use
+    wget -nc http://ftp.acc.umu.se/mirror/gnu.org/savannah/freetype/freetype-2.4.0.tar.gz
+     
+    tar xvf freetype-2.4.0.tar.gz
+    rm freetype-2.4.0.tar.gz
+
+    cd freetype-2.4.0
+
+    if [ $(uname -m) = "ppc64le" ]; then
+      PARAMS="--build=$(rpm --eval %{_host})"
+    fi
+     
+    # We get the files we need at $WORKING_DIR/installedfreetype
+    bash ./configure --prefix=$WORKING_DIR/$OPENJDK_REPO_NAME/installedfreetype $PARAMS && make all && make install
+
+    if [ $? -ne 0 ]; then
+      echo "${error}Failed to configure and build libfreetype, exiting"
+      exit;
+    else
+      echo "${good}Successfully configured OpenJDK with the FreeType library (libfreetype)!"
+    fi
+    echo $normal
+  fi  
+}
+
+checkingAndDownloadCaCerts() 
+{
+  cd $WORKING_DIR
+
+  echo "Retrieving cacerts file"
+
+  # Ensure it's the latest we pull in
+  rm -rf $WORKING_DIR/cacerts_area
+
+  git clone https://github.com/AdoptOpenJDK/openjdk-build.git cacerts_area
+  echo "cacerts should be here..."
+  file $WORKING_DIR/cacerts_area/security/cacerts
 
   if [ $? -ne 0 ]; then
-    echo "${error}Failed to configure and build libfreetype, exiting"
+    echo "Failed to retrieve the cacerts file, exiting..."
     exit;
   else
-    echo "${good}Built the JDK!"
+    echo "${good}Successfully retrieved the cacerts file!"
+  fi
+}
+
+downloadingRequiredDependencies() 
+{
+  echo "Downloading required dependencies...: Asla, Freetype, and CaCerts."
+  checkingAndDownloadingAsla
+  checkingAndDownloadingFreetype
+  checkingAndDownloadCaCerts
+}
+
+
+configuringBootJDKConfigureParameter()
+{
+  if [ -z "$JDK_BOOT_DIR" ] ; then
+    echo "JDK_BOOT_DIR is $JDK_BOOT_DIR"
+    JDK_BOOT_DIR=/usr/lib/java-1.7.0
+  else
```

Comment:
`JDK_BOOT_DIR` can already be set via environment variable so this is fine here i think